src/coinbase/smart_wallet.ts

```diff
@@ -0,0 +1,110 @@
+import { NetworkIdentifier, SmartWallet as SmartWalletModel } from "../client";
+
+import { Coinbase } from "./coinbase";
+
+import type { Hex } from 'viem';
+import type {
+  UserOperationCalls
+} from 'viem/account-abstraction';
+
+import { encodeFunctionData, LocalAccount } from "viem";
+import { UserOperation } from "./user_operation";
+
+export class SmartWallet {
+  private model: SmartWalletModel;
+  private account: LocalAccount;
+  private networkId?: NetworkIdentifier;
+
+  public constructor(model: SmartWalletModel, account: LocalAccount) {
+    this.model = model;
+    this.account = account;
+  }
+
+  public static async create({account}: {account: LocalAccount}): Promise<SmartWallet> {
+    const result = await Coinbase.apiClients.smartWallet!.createSmartWallet({
+        owner: account.address,
+    });
+
+    const smartWallet = new SmartWallet(result.data!, account);
+    return smartWallet;
+  }
+
+  public static async connect({smartWalletAddress, account}: {smartWalletAddress: string, account: LocalAccount}): Promise<SmartWallet> {
+    const result = await Coinbase.apiClients.smartWallet!.getSmartWallet(smartWalletAddress);
+
+    if (!result.data?.owners.some(owner => owner.toLowerCase() === account.address.toLowerCase())) {
+      throw new Error('Account is not an owner of the smart wallet');
+    }
+
+    const smartWallet = new SmartWallet(result.data!, account);
+    return smartWallet;
+  }
+
+  public async use({networkId}: {networkId: NetworkIdentifier}) {
```

Comment:
Similar comment here, although it is more parseable. Weirdly, I would probably name this method `connect` in a vacuum (i.e. the SmartWallet is connecting to a network).

Reply:
Imo explicit is better, `useNetwork`. We're not actually connecting to anything locally, we're setting the network of operation